src/pages/workspaces/[workspaceSlug]/settings.tsx

```diff
@@ -185,6 +185,15 @@ export const getServerSideProps = createGetServerSideProps({
         notFound: true,
       };
     }
+    const { permissions } = data.workspace;
+    if (!permissions.manageMembers) {
```

Comment:
IMO it's the same case as the `notFound`. Better to display a 404 than redirecting the user to a random page